### PR TITLE
[pytorch][build][test] Fix PT1.10.2 mpi test and CW logs

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
This PR skips the MPI tests for the images less than PT1.11. It also fixes the CW logging by adding the `ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main` line to inference images.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.10/py3/Dockerfile.e3.cpu [hunk 1]
```diff
@@ -13,6 +13,7 @@ ARG PT_TORCHVISION_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/pyt
 ENV LANG C.UTF-8
 ENV LD_LIBRARY_PATH /opt/conda/lib/:$LD_LIBRARY_PATH
 ENV PATH /opt/conda/bin:$PATH
+ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main
```
Comment (saimidu) [inline pytorch/inference/docker/1.10/py3/Dockerfile.e3.cpu:16]: This needs to be added into the SageMaker serving dockerfile, not the E3 dockerfile.

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py [hunk 2]
```diff
@@ -189,6 +189,9 @@ def test_pytorch_mpi_gpu(pytorch_training, ec2_connection, gpu_only, py3_only, e
     """
     if test_utils.is_image_incompatible_with_instance_type(pytorch_training, ec2_instance_type):
         pytest.skip(f"Image {pytorch_training} is incompatible with instance type {ec2_instance_type}")
+    _, framework_version = test_utils.get_framework_and_version_from_tag(pytorch_training)
+    if Version(framework_version) < Version("1.11.0"):
+        pytest.skip(f"MPI tests are skipped for PT<1.11 images")
```
Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:194]: Please use fixture `pt111_and_above_only` instead of putting the skip as part of the test. By using the `pt111_and_above_only` to prevent the test function from executing, we can prevent test resources, such as EC2 instances, from being created.

Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:194]: Please use fixture `pt111_and_above_only` instead of putting the skip as part of the test. By using the `pt111_and_above_only` to prevent the test function from executing, we can prevent test resources, such as EC2 instances, from being created.

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py [hunk 3]
```diff
@@ -201,6 +204,9 @@ def test_pytorch_mpi_cpu(pytorch_training, ec2_connection, cpu_only, py3_only, e
     """
     Tests mpi backend
     """
+    _, framework_version = test_utils.get_framework_and_version_from_tag(pytorch_training)
+    if Version(framework_version) < Version("1.11.0"):
+        pytest.skip(f"MPI tests are skipped for PT<1.11 images")
```
Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:209]: Please use fixture `pt111_and_above_only` instead of putting the skip as part of the test. By using the `pt111_and_above_only` to prevent the test function from executing, we can prevent test resources, such as EC2 instances, from being created.

Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:209]: Please use fixture `pt111_and_above_only` instead of putting the skip as part of the test. By using the `pt111_and_above_only` to prevent the test function from executing, we can prevent test resources, such as EC2 instances, from being created.

File: test/dlc_tests/sanity/test_anaconda.py [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
+from packaging.version import Version
 
 @pytest.mark.usefixtures("sagemaker")
 @pytest.mark.model("N/A")
 @pytest.mark.integration("anaconda_removal")
 def test_repo_anaconda_not_present(image):
     """Test to see if all packages installed in the image do not come from repo.anaconda.com"""
+    framework, framework_version = test_utils.get_framework_and_version_from_tag(image)
+    if framework == "pytorch" and Version(framework_version) < Version("1.11.0"):
+        pytest.skip("Skipping for pytorch versions < 1.11")
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_anaconda.py:13]: Let's not have this here, because we want to be able to detect that this test fails, and then remove these packages.

We can make an exception for the pipelines and override the failure of this test, if it occurs.

Comment (saimidu) [inline test/dlc_tests/sanity/test_anaconda.py:13]: Let's not have this here, because we want to be able to detect that this test fails, and then remove these packages.

We can make an exception for the pipelines and override the failure of this test, if it occurs.

**Review by saimidu — APPROVED** [review]

**Review by saimidu — APPROVED** [review]
